index.js

```diff
@@ -3,325 +3,344 @@
+		cameraStyle: PropTypes.any,
+		markerStyle: PropTypes.any,
+		topViewStyle: PropTypes.any,
+		bottomViewStyle: PropTypes.any,
+		topContent: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
+		bottomContent: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
+		notAuthorizedView: PropTypes.element,
+		permissionDialogTitle: PropTypes.string,
+		permissionDialogMessage: PropTypes.string,
+		buttonPositive: PropTypes.string,
+		checkAndroid6Permissions: PropTypes.bool,
+		flashMode: PropTypes.oneOf(CAMERA_FLASH_MODES),
+		cameraProps: PropTypes.object,
+		cameraTimeoutView: PropTypes.element,
+	};
 
-  static defaultProps = {
-    onRead: () => console.log('QR code scanned!'),
-    reactivate: false,
-    vibrate: true,
-    reactivateTimeout: 0,
-    fadeIn: true,
-    showMarker: false,
-    cameraType: 'back',
-    notAuthorizedView: (
-      <View
-        style={{
-          flex: 1,
-          alignItems: 'center',
-          justifyContent: 'center'
-        }}
-      >
-        <Text
-          style={{
-            textAlign: 'center',
-            fontSize: 16
-          }}
-        >
-          Camera not authorized
-        </Text>
-      </View>
-    ),
-    pendingAuthorizationView: (
-      <View
-        style={{
-          flex: 1,
-          alignItems: 'center',
-          justifyContent: 'center'
-        }}
-      >
-        <Text
-          style={{
-            textAlign: 'center',
-            fontSize: 16
-          }}
-        >
-          ...
-        </Text>
-      </View>
-    ),
-    permissionDialogTitle: 'Info',
-    permissionDialogMessage: 'Need camera permission',
-    buttonPositive: 'OK',
-    checkAndroid6Permissions: false,
-    flashMode: CAMERA_FLASH_MODE.auto,
-    cameraProps: {}
-  };
+	static defaultProps = {
+		onRead: () => console.log('QR code scanned!'),
+		reactivate: false,
+		vibrate: true,
+		reactivateTimeout: 0,
+		cameraTimeout: 0,
+		fadeIn: true,
+		showMarker: false,
+		cameraType: 'back',
+		notAuthorizedView: (
+			<View
+				style={{
+					flex: 1,
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+			>
+				<Text
+					style={{
+						textAlign: 'center',
+						fontSize: 16,
+					}}
+				>
+					Camera not authorized
+				</Text>
+			</View>
+		),
+		pendingAuthorizationView: (
+			<View
+				style={{
+					flex: 1,
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+			>
+				<Text
+					style={{
+						textAlign: 'center',
+						fontSize: 16,
+					}}
+				>
+					...
+				</Text>
+			</View>
+		),
+		permissionDialogTitle: 'Info',
+		permissionDialogMessage: 'Need camera permission',
+		buttonPositive: 'OK',
+		checkAndroid6Permissions: false,
+		flashMode: CAMERA_FLASH_MODE.auto,
+		cameraProps: {},
+		cameraTimeoutView: (
+			<View
+				style={{
+					flex: 0,
+					alignItems: 'center',
+					justifyContent: 'center',
+					backgroundColor: 'transparent',
+					height: Dimensions.get('window').width,
```

Comment:
is this right as `Dimensions.get('window').width,`? I mean, it shouldn't be like:

```suggestion
					height: Dimensions.get('window').height,
```

Reply:
here I tried to copy the styles.camera because the camera timeout will replace the Camera element. In styles.camera it uses Dimensions.get('window').width as its height. So I followed it.

```js
camera: {
		flex: 0,
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: 'transparent',
		height: Dimensions.get('window').width,
		width: Dimensions.get('window').width,
	}
```

Reply:
fair enough, I've tested and it seems to be ok